scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/TaskKiller.java

```diff
@@ -67,11 +69,26 @@ private void interruptRepeatedly() {
         }
     }
 
+    /**
+     * Interrupts and gives the thread time to cleanup
+     */
+    private void interruptAndWaitCleanupTime() {
+        threadToKill.interrupt();
+        int iterationsToHitWaitingTimeLimit = CLEANUP_SECONDS_AFTER_FIRST_INTERRUPT * 100;
+        for(int i = 0; threadToKill.isAlive() && i < iterationsToHitWaitingTimeLimit ; i++ ) {
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+            }
+        }
+    }
+
     public void kill(Status status) {
         synchronized (this) {
             wasKilled = true;
             this.status = status;
         }
+        interruptAndWaitCleanupTime();
```

Comment:
Why not using your `interruptRepeatedly` only with the call to `threadToKill.interrupt();` just before? I don't understand why you use two loops successively that seem to have the same purpose?